### ISSUE TITLE
Create a VueJS option for the SPA Modules

### ISSUE THREAD

**Issue by mtrutledge** [mtrutledge/generator-dnn#36]
During the Southern Fried DNN call on 7/19/2018 a suggestion was made to have VueJS as a option for the SPA modules. I think this is a great idea!

**Comment by ralphwilliams** [comment]
I'll be happy to help with this. I will just need some help getting the project set up.

**Comment by mtrutledge** [comment]
@ralphwilliams I have put out some contribution documentation and I am moving items around in the SPA/templates directory structure so you can add the VueJS template easily. I will post more once I have the development branch updated with the directory structure changes.

**Comment by mtrutledge** [comment]
Ok I have updated the development branch to include a new structure under the SPA folder. A yeoman generator lives under the generators folder and has a template. Once you select a generator it runs index.js. So during the initial prompt when I ask the user what type of module they want to generate when they select SPA the index.js in generator-dnn/generators/spa/ is executed. If you look on the development branch I added some placeholder prompts in generator-dnn/generators/spa/index.js. ReactJS is active and VueJS is currently disabled. Essentially you will add a new folder called VueJS under generator-dnn/generators/spa/ and then will put all the necessary template files in there. You will then need to alter generator-dnn/generators/spa/index.js to copy the necessary files.

**Comment by nvisionative** [comment]
@mtrutledge can we do the same thing for Angular?  See #41

**Comment by mtrutledge** [comment]
Absolutely

**Comment by wesfincher** [comment]
Curious if any movement has been made on the VueJs addition?  There's a number of forks that I've found so unsure which is the most recent iteration of this repo.

**Comment by mtrutledge** [comment]
I haven't worked on this tool in a while. I would like to get back to it, but I wasn't really receiving any pull requests from the community either, so I wasn't sure how effective or useful this tool was.

**Comment by wesfincher** [comment]
Yeah, I imagine that kills one's motivation...I only recently discovered yeoman so I'm not sure how I could help out at this point.  I'll give it a go locally.

**Comment by david-poindexter** [comment]
@wesfincher in addition to this tool, you may want to also check out https://github.com/UpendoVentures/generator-upendodnn (which is actively maintained).  It follows a different project structure, but maybe that will fit your preferences as well.  

Lastly, there is a fantastic Visual Studio template that would also accommodate a VueJS module.  This is honestly our "go to" solution these days for creating DNN modules.  https://github.com/valadas/Eraware_Dnn_Templates

**Comment by wesfincher** [comment]
@david-poindexter - Thanks for the advice.  I tried the upendodnn but the VueJS option is disabled when I use the npm package :-(

Unrelated, how are folks doing Continuous Deployment in DNN?  We've been using the PolyDeploy to deploy multiple modules at once, and have looked at (but not tried) the DeployClient (https://github.com/DNNCommunity/PolyDeploy/tree/develop/DeployClient).  If we could get to true CD I would be forever grateful.

**Comment by david-poindexter** [comment]
@wesfincher you are welcome.  You may want to check out https://dnncommunity.org/forums/aft/79 regarding the use of PolyDeploy for CI/CD.

**Comment by mtrutledge** [comment]
@wesfincher I use PolyDeploy and the DeployClient to deploy the packages. I
use Octopus Deploy to coordinate the deployments between my environments.

**Comment by valadas** [comment]
If you like https://github.com/valadas/Eraware_Dnn_Templates and need a had getting Vue working in it, feel free to ping me. I have used Vue once to give it a try but it should just be a matter of getting the starter template in the Module.Web replacing most of what is there.

And yeah, I have been using Polydeploy too, when I have some time I would like to revamp some of the UI there and make the command-line a dotnet-tool on nuget to help even more with CI/CD pipelines. Right now you need to have the cli executable committed in the repo... But I do have a working pipeline if you need help with that too...
